pkg/scripts/download.go

```diff
@@ -51,6 +40,50 @@ func NewDownloader() Downloader {
 	}
 }
 
+// DownloadFile downloads the file pointed to by URL into local targetFile
+// Returns information a boolean flag informing whether any download/copy operation
+// happened and an error if there was a problem during that operation
+func (d *downloader) Download(url *url.URL, targetFile string) (bool, error) {
+	sr := d.schemeReaders[url.Scheme]
+
+	if sr == nil {
+		glog.Errorf("No URL handler found for %s", url.String())
+		return false, stierr.NewURLHandlerError(url.String())
+	}
+	if sr.IsFromImage() {
+		glog.V(2).Infof("Using image internal scripts from: %s", url.String())
+		return false, nil
+	}
+	reader, err := sr.Read(url)
+	if err != nil {
+		glog.Warningf("Unable to download %s (%s)", url.String(), err)
+		return false, err
+	}
+	defer reader.Close()
+
+	out, err := os.Create(targetFile)
+	defer out.Close()
+
+	if err != nil {
+		defer os.Remove(targetFile)
```

Comment:
I don't see the point of doing this in defer if you are going to return in next line.